week2/project/hackyourweather/src/components/weather.js

```diff
@@ -0,0 +1,51 @@
+import React, { useState } from 'react';
+import CityName from './CityName'
+import Info from './WeatherInfo'
+import Temp from './Temp'
+import Form from './Form'
+
+function Weather() {
+    const [weatherInfo, setWeatherInfo] = useState({})
+    const [loading, setLoading] = useState(false)
+    const [wrong, setWrong] = useState(true)
+
+    const getCity = async (e) => {
+        e.preventDefault();
+        const city = e.target.elements.cityName.value
+
+        try {
+            const res = await fetch(`http://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${process.env.REACT_APP_OPENWEATHERMAP_API_KEY}`)
+            if (res.ok){
+                const data = await res.json()
+                setWeatherInfo(data)
+                setLoading(true)
+                setWrong(true)
+            }else{
+                setWrong(false)
+                throw new Error();
```

Comment:
`setLoading` and do we have any error message? may be:
```suggestion
                throw new Error('Oops something went wrong!');
```